src/redux/rockets/rockets.js

```diff
@@ -0,0 +1,67 @@
+import axios from 'axios';
+
+const FETCH_ROCKETS = 'App/rockets/GET_ROCKETS';
+const RESERVE_ROCKET = 'App/rockets/RESERVE_ROCKET';
+const CANCEL_ROCKET_RESERVATION = 'App/rockets/CANCEL_ROCKET_RESERVATION';
+const url = 'https://api.spacexdata.com/v3/rockets';
+
+export const fetchRockets = (payload) => ({
+  type: FETCH_ROCKETS,
+  payload,
+});
+
+export const getRockets = () => async (dispatch) => {
+  const { data } = await axios.get(url);
+  const rockets = [];
+  data.forEach((rocket) => {
+    const newRocket = {
+      id: rocket.id,
+      name: rocket.rocket_name,
+      description: rocket.description,
+      image: rocket.flickr_images[0],
+      reserved: false,
+    };
+    rockets.push(newRocket);
+  });
+  dispatch(fetchRockets(rockets));
+};
+
+export const reserveRocket = (id) => async (dispatch) => {
+  dispatch({ type: RESERVE_ROCKET, payload: id });
+};
+
+export const cancelRocket = (id) => async (dispatch) => {
+  dispatch({ type: CANCEL_ROCKET_RESERVATION, payload: id });
+};
```

Comment:
- [ ] Here you are using an unnecessary async function for those types of actions (`synchronous actions`). Note that these actions are not blocking, the payload that they are sending to the reducer function is already there, so you don't need to make them asynchronous. They will be dispatched in your component, e.g `dispatch(reserveRocket(rocketId)`

- [ ] Let's keep them simple like this
```
export const reserveRocket = (id) => ({ type: RESERVE_ROCKET, payload: id });

export const cancelRocket = (id) => ({ type: CANCEL_ROCKET_RESERVATION, payload: id });
```